src/game_battler.cpp

```diff
@@ -361,10 +361,36 @@ bool Game_Battler::AddState(int state_id, bool allow_battle_states) {
 	return was_added;
 }
 
+void ForEachStateRemoved(Game_Battler& battler, Span<int16_t> removed_states) {
+	for (const auto& state_id: removed_states) {
+		auto prev_restriction = battler.GetSignificantRestriction();
+
+		if (state_id == lcf::rpg::State::kDeathID) {
+			battler.SetHp(1);
+		}
+
+		auto cur_restriction = battler.GetSignificantRestriction();
```

Comment:
I'm probably missing something but doesn't this always make `prev_restriction` and `cur_restriction` both equal?

Reply:
Yes this is a bug